build.py

```diff
@@ -0,0 +1,357 @@
+#!/usr/bin/env python
+
+# Keep this file compatible with Python 2 -- it's the first file users will run.
+# Even if we end up requiring a different version for the rest of the project,
+# this file should probably not crash and burn.
+
+import argparse
+import errno
+import itertools
+import os
+import subprocess
+import sys
+import traceback
+
+try: import ctypes  # necessary for Windows
+except ImportError: ctypes = None
+
+pip = None  # Will be lazily imported if needed (lazy because the rest of this script should run even on installations without pip)
+
+StringishTypes = (str, bytes) if sys.version_info[0] >= 3 else (basestring, str, unicode)
+
+def pip_main(*args):
+    global pip  # Don't import multiple times
+    if pip is None: import pip
+    main = getattr(pip, 'main', None)
+    if not main:
+        # This is technically unsupported... https://stackoverflow.com/a/50255019
+        # However, calling pip _externally_ for an interpreter that is _already_ running
+        # doesn't seem like any less fragile a scenario!
+        from pip._internal import main
+    return main(*args)
+
+def uniquify(items):
```

Comment:
I don't think we need this. If we really do need it, we can do `list(set(items))`